core/templates/filters/string-utility-filters/camel-case-to-hyphens.pipe.ts

```diff
@@ -23,9 +23,9 @@ import { Injectable, Pipe, PipeTransform } from '@angular/core';
 })
 @Pipe({name: 'camelCaseToHyphens'})
 export class CamelCaseToHyphensPipe implements PipeTransform {
-  transform(input: string): string {
-    var result = input.replace(/([a-z])?([A-Z])/g, '$1-$2').toLowerCase();
-    if (result[0] === '-') {
+  transform(input: string | null): string | undefined {
+    var result = input?.replace(/([a-z])?([A-Z])/g, '$1-$2').toLowerCase();
+    if (result && result[0] === '-') {
```

Comment:
Can you explain these changes?

Reply:
Since we are making interactionId to be potentially null, the transform function is used here (https://github.com/oppia/oppia/blob/develop/core/templates/services/exploration-html-formatter.service.ts#L169). As a result, we have to set the input to transform function to null as well, otherwise running strict checks lead to the error:  Argument of type 'string | null' is not assignable to parameter of type 'string'.

Reply:
Can we not allow null here now that we throw an error in HTML formatter?

Reply:
Yep, I have removed them now.